easybuild/easyblocks/g/gromacs.py

```diff
@@ -123,6 +124,16 @@ def configure_step(self):
             # disable GUI tools
             self.cfg.update('configopts', "-DGMX_X11=OFF")
 
+            # convince to build for an older architecture then present on the build node by setting GMX_SIMD CMake flag.
+            if build_option('optarch', default=None) and self.toolchain.comp_family() in build_option('optarch'):
```

Comment:
You should also check here whether `optarch` is specified by compiler?

In theory, you could also have a hit here with `GCC` when `optarch` is just specified as a set of options (regardless of compiler), which would make this crash and burn below?

You could use something like:
```python
if isinstance(optarch, dict) and ...
```